src/app/components/paginator/index.tsx

```diff
@@ -0,0 +1,73 @@
+import { faAngleLeft, faAngleRight } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { AVAILABLE_PAGES_RANGE, PAGE_PARAM, PAGINATION_LIMIT } from 'app/constants';
+import { Button } from 'atomic/atm.button';
+import Loading from 'react-loading';
+import { useSearchParams } from 'react-router-dom';
+
+interface PaginatorProps {
+  isLoading: boolean;
+  count: number;
+  limit: number;
+}
+
+export function getCurrentPage(searchParams: URLSearchParams): number {
+  const currentPageString = searchParams.get(PAGE_PARAM);
+  return currentPageString ? parseInt(currentPageString) : 1;
+}
+
+export function getCurrentOffset(searchParams: URLSearchParams): number {
+  return getCurrentPage(searchParams) * PAGINATION_LIMIT;
+}
+
+export function Paginator({ isLoading, count, limit }: PaginatorProps) {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const currentPage = getCurrentPage(searchParams);
+  const firstPage = 1;
+  const lastPage = Math.floor(count / limit);
+
+  const getAvailablePages = (currentPage: number) => {
```

Comment:
essa função serve para pegar as páginas né? digo, se tem 10 páginas ela gera um array assim -> [1, 2, 3, 4, ..., 10]
se for isso, eu acho que o server retorna a página atual e o total de páginas, então não precisa 'calcular' aqui - supondo que o total de páginas seja `pagesCount`, ficaria algo assim:
```ts
const getAvailablePages = () => Array.from({length: 10}, (_, i) => i + 1);
// [1, 2, 3, 4, 5, ..., pageCount]
// source: https://stackoverflow.com/a/33352604/12393495
```

Reply:
se existirem 20 paginas e atualmente estamos na pagina 7, essa função retornaria um array assim: [1, 5, 6, 7, 8, 9, 20]. Com base nesse array eu coloco as opções de paginas (2 anteriores + 2 posteriores + primeira e ultima).
Esqueci de anexar a imagem, mas ta ai:
![pagination](https://user-images.githubusercontent.com/13165470/158658037-747ad260-15a7-438b-965b-2f565d511d6c.png)
:

Reply:
lembrando que ainda não tem estilização, por isso o visual ta bem zoadinho